lex.go

```diff
@@ -414,7 +410,11 @@ func lexValue(lx *lexer) stateFn {
 	case r == '.': // special error case, be kind to users
 		return lx.errorf("Floats must start with a digit, not '.'.")
 	}
-	return lx.errorf("Expected value but found %q instead.", r)
+
+	// Assume barewords are constants so we can emit better error
+	// messages.
+	lx.backup()
+	return lexBareword
```

Comment:
I prefer how the code worked before, by only looking for a boolean when parsing a value beginning with `t` or `f`.

Instead of introducing a catch-all code path which ends up only handling booleans, it shouldn't be hard to instead change `lexConst` to give the improved error message by parsing a whole "constant" and comparing it to the target string.